File: backend/app/models/event.py
from datetime import datetime, timezone
from sqlalchemy import Column, Integer, String, Text, Time, Date, Boolean, DateTime, Enum, SmallInteger, ForeignKey
from sqlalchemy.orm import relationship

from app.database import Base


class FixedEvent(Base):
    __tablename__ = "fixed_events"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)

    title = Column(String(200), nullable=False)
    description = Column(Text)

    day_of_week = Column(SmallInteger, comment="0=Sun .. 6=Sat")
    start_time = Column(Time, nullable=False)
    end_time = Column(Time, nullable=False)

    start_date = Column(Date)
    end_date = Column(Date)
    is_recurring = Column(Boolean, default=False)
    recurrence_rule = Column(String(255))

    category = Column(Enum("college", "work", "part_time", "appointment", "other", name="event_category"), nullable=False, default="other")
    color = Column(String(7), default="#3788d8")
    is_active = Column(Boolean, default=True)

    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), nullable=False)

    user = relationship("User", back_populates="fixed_events")

File: mobile/lib/models/event.dart
class FixedEvent {
  final int id;
  final int userId;
  final String title;
  final String? description;
  final int? dayOfWeek;
  final String startTime;
  final String endTime;
  final bool isRecurring;
  final String category;
  final String? color;

  FixedEvent({
    required this.id,
    required this.userId,
    required this.title,
    this.description,
    this.dayOfWeek,
    required this.startTime,
    required this.endTime,
    required this.isRecurring,
    required this.category,
    this.color,
  });

  factory FixedEvent.fromJson(Map<String, dynamic> json) => FixedEvent(
    id: json['id'] as int,
    userId: json['user_id'] as int,
    title: json['title'] as String,
    description: json['description'] as String?,
    dayOfWeek: json['day_of_week'] as int?,
    startTime: json['start_time'] as String,
    endTime: json['end_time'] as String,
    isRecurring: json['is_recurring'] as bool? ?? false,
    category: json['category'] as String,
    color: json['color'] as String?,
  );

  Map<String, dynamic> toJson() => {
    'title': title,
    'description': description,
    'day_of_week': dayOfWeek,
    'start_time': startTime,
    'end_time': endTime,
    'is_recurring': isRecurring,
    'category': category,
    'color': color,
  };
}
